addon/components/stacked-vertical-bar-chart.js

```diff
@@ -0,0 +1,1015 @@
+import Ember from 'ember';
+import ChartComponent from './chart-component';
+import LegendMixin from '../mixins/legend';
+import FloatingTooltipMixin from '../mixins/floating-tooltip';
+import AxesMixin from '../mixins/axes';
+import FormattableMixin from '../mixins/formattable';
+import NoMarginChartMixin from '../mixins/no-margin-chart';
+import AxisTitlesMixin from '../mixins/axis-titles';
+import LabelTrimmer from '../utils/label-trimmer';
+
+/**
+ * Base class for stacked vertical bar chart components.
+ *
+ * Supersedes the deprecated functionality of VerticalBarChartComponent
+ * with stackBars: true.
+ * @class
+ * @augments ChartComponent
+ */
+const StackedVerticalBarChartComponent = ChartComponent.extend(LegendMixin,
+  FloatingTooltipMixin, AxesMixin, FormattableMixin, NoMarginChartMixin,
+  AxisTitlesMixin, {
+
+  classNames: ['chart-vertical-bar', 'chart-stacked-vertical-bar'],
+
+  // ---------------------------------------------------------------------------
+  // Stacked Vertical Bar Chart Options
+  // ---------------------------------------------------------------------------
+
+  /**
+   * The smallest slices will be combined into an 'Other' slice until no slice
+   * is smaller than minSlicePercent.
+   * @type {number}
+   */
+  minSlicePercent: 2,
+
+  /**
+   * Data without a barLabel will be merged into a bar with this name
+   * @type {string}
+   */
+  ungroupedSeriesName: 'Other',
+
+  /**
+   * The maximum number of slices. If the number of slices is greater
+   * than this, the smallest slices will be combined into an 'Other' slice until
+   * there are at most maxNumberOfSlices (including the 'Other' slice).
+   * @type {number}
+   */
+  maxNumberOfSlices: 10,
+
+  /**
+   * If there are more slice labels than maxNumberOfSlices and/or if there are
+   * slice types that do not meet the `minSlicePercent`, the smallest slices
+   * will be aggregated into an 'Other' slice. This property defines the label
+   * for this aggregate slice.
+   * @type {string}
+   */
+  otherSliceLabel: 'Other',
+
+  /**
+   * Width of slice outline, in pixels
+   * @type {number}
+   */
+  strokeWidth: 1,
+
+  /**
+   * Space between bars, as fraction of bar size
+   * @type {number}
+   */
+  betweenBarPadding: Ember.computed('numSlices', function() {
+    // Use padding to make sure bars have a maximum thickness.
+    var scale = d3.scale.linear().domain([1, 8]).range([1.25, 0.25]).clamp(true);
+    return scale(this.get('numSlices'));
```

Comment:
You raise a good point here, and I want to give some background. This code was copied over from the vertical bar chart and we largely did not change it because we did not modify how bars are spaced, just slices within the bars. That said, this is less than ideal and we might as well correct it in this file.

First off, you're right -- this should be a fixed number that is set by the developer. It is only used in `xBetweenBarScale`, and sets the inner padding and outer padding for bars (to the same value). Because of that, it should only range between 0 and 1, where 0 represents NO padding between bars and 1 represents FULL padding between bars (aka, bars with a width of 0). 

By nature, this is a visual setting of the chart, so it should have some logical default value (I'd say either 0 or 0.5), with the option to override. Even in the demo app for ember charts we show a slider that controls this property and sets it to a raw value, meaning that as soon as we touch the slider, this computed property gets chucked out the window in the demo app. If we're not even using the computed property in the demo app, it doesn't make sense to complicate it behind the curtain.

Lastly, though, even with this being a fixed number does mean it still _scales_ with the width of the chart, because it represents what proportion of the chart should be filled with padding. It is not a raw number of pixels. This is expected and desired behavior, though.